lib/uefi/runtime_service_provider.cpp

```diff
@@ -22,12 +22,19 @@
 #include <string.h>
 #include <uefi/types.h>
 
+#include "variable_mem.h"
+
 namespace {
 
 constexpr auto &&kSecureBoot = "SecureBoot";
 
 EFI_STATUS GetVariable(char16_t *VariableName, EfiGuid *VendorGuid,
                        uint32_t *Attributes, size_t *DataSize, void *Data) {
+
+  if (!VariableName || !VendorGuid || !DataSize) {
+    return INVALID_PARAMETER;
+  }
+
   char buffer[512];
```

Comment:
You can do this in a later CL

replace this with the newly added charset functions